internal/cmd/lsp-record/lsp-record.go

```diff
@@ -128,15 +128,23 @@ func retryDial(network, address string) (net.Conn, error) {
 	return conn, err
 }
 
-// massageGitHubArchive rewrites filenames to match what we expect. GitHub
-// archives always have a top-level dir, so strip it out.
+// massageGitHubArchive rewrites filenames to match what we expect and removes
+// directory entries. GitHub archives always have a top-level dir, so strip it
+// out.
 //
 // before: dockerfile-language-server-nodejs-3083f51108b5e5ddfd440e6fe3da415d10b9c69c/src/server.ts
 // after:  /src/server.ts
 func massageGitHubArchive(r *zip.ReadCloser) {
-	for i, file := range r.File {
-		r.File[i].Name = file.Name[strings.Index(file.Name, "/"):]
+	i := 0
+	for _, file := range r.File {
+		if file.FileInfo().IsDir() {
```

Comment:
Is the only reason that we're doing this is to match to behavior that Sourcegraph's VFS abstraction has (only return files, not dirs)?

Reply:
yes. But ideally it should be part of the spec as well. Right now some archives returned by github do list directories in the zip file (but not all).